lib/lens/macros.ex

```diff
@@ -13,7 +13,16 @@ defmodule Lens.Macros do
     end
 
     quote do
-      def unquote(header), do: unquote(body)
+      def unquote(header) do
+        lens = unquote(body)
+        fn
+          :get, data, next ->
```

Comment:
Ok, maybe let's finish up this PR first, because it's already a monster, and then I can figure out what to do with these macros. Please address the other comments, I made and then I can merge. I think I came around about the order of arguments to `Lens.map`, etc. `Regex` also has this order even though it's really weird to use, so I guess we're in good company.